src/vuex/modules/index/store.ts

```diff
@@ -1,230 +1,269 @@
   },
-  addIndex(state, index) {
-    Vue.set(state.indexesAndCollections, index, new Index())
+  addIndex(state, index: Index) {
+    state.indexes.push(index)
   },
-  deleteIndex(state, index) {
-    Vue.delete(state.indexesAndCollections, index)
+  removeIndex(state, index: Index) {
+    Vue.delete(state.indexes, getIndexPosition(state.indexes, index.name))
   },
-  removeRealtimeCollection(state, { index, collection }) {
-    if (
-      !state.indexesAndCollections[index] ||
-      !state.indexesAndCollections[index].realtime
-    ) {
-      return
-    }
-
-    // prettier-ignore
-    state.indexesAndCollections[index].realtime =
-      state
-        .indexesAndCollections[index]
-        .realtime
-        .filter(realtimeCollection => realtimeCollection !== collection)
-  },
-  removeStoredCollection(state, { index, collection }) {
-    if (
-      !state.indexesAndCollections[index] ||
-      !state.indexesAndCollections[index].stored
-    ) {
-      return
-    }
-
-    const idx = state.indexesAndCollections[index].stored.indexOf(collection)
-    Vue.delete(state.indexesAndCollections[index].stored, idx)
+  removeIndexes(state, indexes: Index[]) {
+    const keepedIndexes = _.difference(state.indexes, indexes)
+    Vue.set(state, 'indexes', keepedIndexes)
+  },
+  removeCollection(state, { index, collection }: IndexCollectionPayload): void {
+    state.indexes[getIndexPosition(state.indexes, index.name)].removeCollection(
+      collection
+    )
   }
 })
 
 const actions = createActions({
-  async createIndex(context, index: string) {
+  async createIndex(context, name: string) {
     const { commit, rootGetters } = indexActionContext(context)
 
-    await rootGetters.kuzzle.$kuzzle.index.create(index)
+    await rootGetters.kuzzle.$kuzzle.index.create(name)
+    const index = new Index(name)
+
     commit.addIndex(index)
   },
-  async deleteIndex(context, index) {
+  async deleteIndex(context, index: Index) {
     const { commit, rootGetters } = indexActionContext(context)
 
-    await rootGetters.kuzzle.$kuzzle.index.delete(index)
-    removeIndex(index)
-    commit.deleteIndex(index)
+    await rootGetters.kuzzle.$kuzzle.index.delete(index.name)
+
+    commit.removeIndex(index)
   },
-  async listIndexes(context) {
+  async bulkDeleteIndexes(context, indexes: Index[]) {
     const { commit, rootGetters } = indexActionContext(context)
+    const indexesNames = indexes.map(el => el.name)
+
+    await rootGetters.kuzzle.$kuzzle.index.mDelete(indexesNames)
+    commit.removeIndexes(indexes)
+  },
+  async fetchIndexList(context) {
+    const { commit, rootGetters } = indexActionContext(context)
+    const indexes: Index[] = []
     commit.setLoadingIndexes(true)
+
     let result = await rootGetters.kuzzle.$kuzzle.index.list()
-    result = result.filter(index => index !== '%kuzzle')
+    result = result.filter(
+      (indexName: string) => indexName !== INTERNAL_INDEX_NAME
```

Comment:
I was wondering if this was still relevant... Kuzzle does not return the internal indexes anymore, does it?